File: erlang/06-io-operations/write_file.erl
#!/usr/bin/env escript
%% write_file.erl - Writing data to a file
main(_) ->
    %% A list of strings is valid iodata - written in one shot
    Lines = ["Erlang 1986\n", "Elixir 2011\n", "Gleam 2016\n"],
    ok = file:write_file("languages.txt", Lines),
    io:format("Wrote languages.txt~n"),

    %% Open in append mode and stream one more line through io:format/3
    {ok, Fd} = file:open("languages.txt", [append]),
    io:format(Fd, "Erlang/OTP 27~n", []),
    file:close(Fd),
    io:format("Appended one line~n").

File: php/06-io-operations/write_file.php
<?php
$lines = [
    "First line",
    "Second line",
    "Third line",
];

// One-shot write: creates the file or overwrites it entirely
file_put_contents("notes.txt", implode("\n", $lines) . "\n");

// Append to an existing file with the FILE_APPEND flag
file_put_contents("notes.txt", "Appended line\n", FILE_APPEND);

// Stream-based writing gives finer control over the open file handle
$handle = fopen("log.txt", "w");
fwrite($handle, "Log entry 1\n");
fwrite($handle, "Log entry 2\n");
fclose($handle);

echo "Files written successfully\n";
